fleetctl/destroy.go

```diff
@@ -33,9 +33,14 @@ Destroyed units are impossible to start unless re-submitted.`,
 }
 
 func runDestroyUnits(args []string) (exit int) {
-	for _, v := range args {
-		name := unitNameMangle(v)
-		err := cAPI.DestroyUnit(name)
+	units, err := findUnits(args)
+	if err != nil {
+		stderr("%v", err)
+		return 1
+	}
+
+	for _, v := range units {
+		err := cAPI.DestroyUnit(v.Name)
```

Comment:
@tixxdz I'm not really happy this got merged without a follow up to this comment (or an LGTM). It's actually a change in behaviour (and does not restore to the semantic of pre- https://github.com/coreos/fleet/commit/9530eedb82ac8a3af7412c62afc5dd30d17d6892 as the commit message implies).

Previously, if we received a command `fleetctl destroy a b c`, and `a` and `b` existed but `c` did not, we would nevertheless successfully remove `a` and `b`. Now, we will immediately error out instead, because `c` couldn't be found.

My suggestion was to just check the response from each `cAPI.DestroyUnit()` call, and ignore the error if it's one indicating the unit doesn't exist (this might require some error plumbing). As it is now, we have a race condition between getting the list of units from `findUnits` and doing the `Destroy` call.